build-tools/rpm/macros.yast

```diff
@@ -67,13 +67,35 @@
         %{__make} %{?jobs:-j%jobs}%{?!jobs:%{?_smp_mflags:%_smp_mflags}} \
     %endif # "x%%{?make_build}" != "x"
 
+# Run the tests, when "--with=yast_run_ci_tests" osc option is set some
+# additional tests are executed (rubocop, spell check...).
 %yast_check \
-    if [ ! -f "%{yast_ydatadir}/devtools/NO_MAKE_CHECK" ]; then \
+    %if %{with yast_run_ci_tests} \
+    if [ -f "configure.in.in" ]; then \
+        # TODO: fix the check:ci task to also work with autotools based modules \
         %{__make} check \\\
             VERBOSE=1 \\\
             Y2DIR="%{buildroot}/%{yast_dir}" \\\
             DESTDIR="%{buildroot}" \
-    fi
+    elif [ -f "Rakefile" ]; then \
+        # TODO: coveralls support \
+        LC_ALL=en_US.utf8 rake --verbose --trace check:ci \
+    fi \
+    %else \
+    if [ ! -f "%{yast_ydatadir}/devtools/NO_MAKE_CHECK" ]; then \
```

Comment:
I think we can remove this NO_MAKE_CHECK feature and move the `%if` inside the `if`